code/game/gamemodes/objective.dm

```diff
@@ -364,11 +408,11 @@
 	return 1
 
 
-/datum/objective/martyr
+/datum/objective/escape_obj/martyr

```

Comment:
Any objective that causes compatibility issues with martyr should be blacklisting it from being rolled.